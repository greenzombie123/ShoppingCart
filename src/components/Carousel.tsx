import style from "./Carousel.module.css";

const Carousel = () => {
  return (
    <div className={style.carouselContainer} data-testid="a">
      <button className={style.leftButton}></button>
      {/* <div className="carousel_wrapper">
                <div className="carousel">
                    <div className="slide"><img src="./assets/dufflebag.webp" alt=""/></div>
                    <div className="slide mainSlide"><img src="./assets/leatherbag.webp" alt=""/>
                        <div className="itemInfo">
                            <div className="name">leather bag</div>
                            <div className="price">$100</div>
                        </div>
                    </div>
                    <div className="slide"><img src="./assets/pinkbag.webp" alt=""/></div>
                    <div className="slide"><img src="./assets/leatherbag.webp" alt=""/></div>
                    <div className="slide"><img src="./assets/pinkbag.webp" alt=""/></div>
                </div>
                <div className="navigation_list">
                    <div><a href=""><img src="./assets/dufflebag.webp" alt=""/></a></div>
                    <div><a href=""><img src="./assets/leatherbag.webp" alt=""/></a></div>
                    <div><a href=""><img src="./assets/pinkbag.webp" alt=""/></a></div>
                    <div><a href=""><img src="./assets/leatherbag.webp"alt=""/></a></div>
                    <div><a href=""><img src="./assets/pinkbag.webp" alt=""/></a></div>
                </div>
            </div> */}
      <button className={style.rightButton}></button>
    </div>
  );
};

export default Carousel;
