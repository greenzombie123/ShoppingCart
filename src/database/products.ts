export type ProductCategory = "Jewelry" | "Men's Clothing" | "Women's Clothing" | "Electronics"

export type Product = {
	id:number,
	name:string,
	price:number,
	rating:number,
	stars:1|2|3|4|5,
	likes:number,
	styles:Style[]
	category:ProductCategory
}

export type Style = {
	description:string,
	picture:URL,
	isCurrentStyle:boolean, 
}

export type CartItem = {
	product:Product,
	id:number,
	quantity:number,
	style:string
}

export type Cart = CartItem[]