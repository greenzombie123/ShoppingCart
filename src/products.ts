// import fs from "fs";

export type ProductCategory =
  | "Jewelry"
  | "Men's Clothing"
  | "Women's Clothing"
  | "Electronics";

export type Product = {
  id: number | string;
  name: string;
  price: number;
  ratings: number;
  stars: 1 | 2 | 3 | 4 | 5;
  likes: number;
  styles: Style[];
  category: ProductCategory;
};

export type Style = {
  description: string;
  picture: string;
  isCurrentStyle: boolean;
};

export type CartItem = {
  name:string,
  price: number;
  id: number;
  quantity: number;
  style: string | undefined,
  picture:string
};

export type Cart = CartItem[];

// const thing: Product[] = [
//   {
//     id: 1,
//     name: "polo shirt",
//     price: 2314,
//     ratings: 334,
//     stars: 5,
//     likes: 380,
//     category: "Men's Clothing",
//     styles: [
//       {
//         description: "black",
//         picture: "/images/poloshirtblack.webp",
//         isCurrentStyle: true,
//       },
//       {
//         description: "blue",
//         picture: "/images/bluepoloshirt.jpg",
//         isCurrentStyle: false,
//       },
//     ],
//   },
//   {
//     id: 1,
//     name: "shirt",
//     price: 1299,
//     ratings: 234,
//     stars: 4,
//     likes: 1000,
//     category: "Men's Clothing",
//     styles: [
//       {
//         description: "white",
//         picture: "src/assets/bluepoloshirt.jpg",
//         isCurrentStyle: true,
//       },
//     ],
//   },
// ];

// const doShit = async () => {
//   const g = await fetch(`http://localhost:3000/test`, {
//     method: "PATCH",
//     headers: { "Content-Type": "application/json" },
//     body:JSON.stringify({test:[]})
//   }).catch(e=>console.log(e))



//   console.log(g)

// };

// doShit()
