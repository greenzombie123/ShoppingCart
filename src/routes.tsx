import { RouteObject } from "react-router-dom";
import App from "./App";
import { addToCart, getCart, getRandomProducts, getStoreItems } from "./Loaders";
import Carousel from "./components/Carousel";
import StorePage from "./components/StorePage";
import ShoppingProduct from "./components/ShoppingProduct";
import ShoppingCart from "./components/ShoppingCart";

const routes: RouteObject[] = [
  {
    path: "/",
    element: <App />,
    loader: getCart,
    children: [
      {
        loader:getRandomProducts,
        element: <Carousel />,
        index:true,
      },
      {
        loader:getStoreItems,
        element:<StorePage/>,
        path:"store/:category"
      },
      {
        action:addToCart,
        element:<ShoppingProduct/>,
        path:"product/:id"
      },
      {
        action:getCart,
        element:<ShoppingCart/>,
        path:"mycart"
      }
    ],
  },
];

export default routes;
