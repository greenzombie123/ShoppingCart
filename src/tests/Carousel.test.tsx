import { describe, expect, it } from "vitest";
import {
  createMemoryRouter,
  MemoryRouter,
  RouteObject,
  RouterProvider,
} from "react-router-dom";
import { render, screen, waitFor } from "@testing-library/react";
import Carousel, { Slider } from "../components/Carousel";
import routes from "../routes";
import { Product } from "../products";
import userEvent from "@testing-library/user-event";

describe("Carousel", () => {
  it("renders buttons on the screen", async () => {
    const router = createMemoryRouter(routes);

    render(<RouterProvider router={router} />);

    const buttons = await waitFor(() => screen.getAllByRole("button"));

    expect(buttons[0]).toBeInTheDocument();
    expect(buttons[1]).toBeInTheDocument();
  });

  it("renders all 5 slides", () => {
    const products: Product[] = [
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 2,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
    ];

    render(
      <MemoryRouter>
        <Slider products={products} />
      </MemoryRouter>
    );

    const slides = screen.getAllByTestId("slide");

    expect(slides[0]).toBeInTheDocument();
    expect(slides[1]).toBeInTheDocument();
    expect(slides[2]).toBeInTheDocument();
    expect(slides[3]).toBeInTheDocument();
    expect(slides[4]).toBeInTheDocument();
    expect(slides[0]).toMatchSnapshot();
  });

  it("Moves the slider to the right when a button is pushed", async () => {
    const products: Product[] = [
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 2,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
      {
        id: 1,
        name: "polo shirt",
        price: 2314,
        ratings: 334,
        stars: 5,
        likes: 380,
        category: "Men's Clothing",
        styles: [
          {
            description: "black",
            picture: "/images/poloshirtblack.webp",
            isCurrentStyle: true,
          },
          {
            description: "blue",
            picture: "/images/bluepoloshirt.jpg",
            isCurrentStyle: false,
          },
        ],
      },
    ];

    const routes:RouteObject[] = [{path:"/", element:<Carousel/>, loader: ()=> products}]

    const user = userEvent.setup()

    const router = createMemoryRouter(routes);

    render(<RouterProvider router={router} />);

    const buttons = await waitFor(()=>screen.getAllByRole<HTMLButtonElement>("button"));

    await user.click(buttons[1])

    const slides = screen.getAllByTestId("slide");

    expect(slides[1]).toHaveClass("main")
  });
});
