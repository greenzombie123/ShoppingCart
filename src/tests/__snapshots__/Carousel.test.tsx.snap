// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > renders buttons on the screen 1`] = `
<div
  class="_carouselContainer_883aa8"
  data-testid="a"
>
  <button
    class="_leftButton_883aa8"
  />
  <button
    class="_rightButton_883aa8"
  />
</div>
`;
